### ISSUE TITLE
Not accessible with the Orca screenreader

### ISSUE THREAD

**Issue by miksuh** [andyholmes/gnome-shell-extension-gsconnect#865]
Hello

First of all this is a nice extension. But unfortunately GSConnect currently is not really accessible with the Orca screen reader. I am blind and I am using GSConnect but e.g the SMS messaging feature is completely unusable with the Orca screen reader. Incoming SMS notifications show just fine in the Gnome shell notification area, but the messaging window is completely inaccessible with the Orca screen reader. Also e.g the extension settings window is not accessible with the Orca screen reader. I am sure that I am not the only blind user who would like to use the GSConnect so please make all parts of the extension accessible with the Orca screen reader.

Please note that blind and other visually impaired users have to use computer from keyboard only without using mouse. You really can not use mouse if you can not see. It should be possible to access all parts of the GSConnect user interface by using keyboard only without using mouse at all. It should be possible to navigate the window's user interface by using TAB -key, arrow keys and possibly maybe also the F6 key. It should also be possible to make selections and click buttons etc by using enter or spacebar. When you navigate the user interface by using tab and arrow keys Orca screen reader should always speak the active button, list item etc. So every button, slider etc should have a text label or text desccription which the Orca screen reader can speak. So ieven if the button shows only an icon on screen the button should still have label which the Orca screen reader can speak.

Orca screen reader should be available in every modern distro. Here is how you can install it on Debian if Orca is not yet installed:

apt install orca

**Comment by andyholmes** [comment]
Thanks, there's an open issue #240 with a checklist that needs to be run through. Unfortunately I don't have much time for this project currently, so help would be appreciated working through that list.

**Comment by ferdnyc** [comment]
If nobody else is able to pick it up, I'll try to get to it some time this week.

**Comment by miksuh** [comment]
Hi


That would be really great. If you could do that it would be greatly 
appreciated.




Frank Dana kirjoitti 24.5.2020 klo 19.49:

**Comment by andyholmes** [comment]
I really intended to do this a while back, but it's bit intensive task especially because the messaging dialog is already complicated for non-accessibility (auto-scrolling and stuff like that).

All excuses aside, I think for the project it should have it get done by GNOME 3.38 at the latest. I think that this is basic functionality and its not fair to keep pretending it isn't.

**Comment by daniellandau** [comment]
@miksuh as you can read above, @andyholmes has been chipping away at this in August and as far as I can tell as a sighted person most of the functionality is there already. At this point I think it would make most sense to test it out and post here any bugs found or other general ways in which things could be better. I'll start:

- [ ] You can select mount in the system menu and then go to the menu again and select files, but when the "files" menu opens, it doesn't have focus or otherwise can't be navigated.
- [ ] Remote keyboard doesn't read out the "remote keyboard not active" warning on first open
- [ ] I couldn't find a way to escape the remote keyboard window without the mouse, as it seems to grab focus and keystrokes very effectively. I don't know what the proper solution would be, but to my uninformed opinion the remote keyboard would seem to be one of the best features for the visually impaired so figuring this out I think should be high priority.

**Comment by ferdnyc** [comment]
Unpinning this, as we need room for other pinned issues. Leaving it open, though, as work is still ongoing.
